src/binding.rs

```diff
@@ -0,0 +1,238 @@
+pub mod mod_keys;
+pub mod relationship;
+
+use core::fmt::{self, Display, Formatter};
+
+use bevy::{
+    ecs::{component::HookContext, world::DeferredWorld},
+    prelude::*,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::prelude::*;
+
+/// Input from any source associated with [`Action<C>`].
```

Comment:
This is a weird definition. A binding to me is "the mapping between actions and inputs". This definition suggests it... statefulyl tracks the current input?